### PR TITLE
`hook_tool` now works with conda envs and venvs

### PR DESCRIPTION
**Pull request by danfulton**
I have been using a conda environment for a lot of development, but hooks took some effort to get working, and are easy to break when switching nodes or re-pulling code, so I have made some effort to automate the hook configuration installation.

The process of installing hook/config files has been moved into the python package installation, and I made modifications to `hook_tool` so that modules can be written more portably by support relative paths.  I have tested that this successfully installs and configures in a fresh user conda environment on perlmutter, and will successfully use hooks.  See the updated `README.md` for installation instructions.

I also modified the RPM build process to use the new python install procedure, and verified that the RPM will build correctly, but did not verify installation/run for RPM build.

Hopefully this makes testing/development a bit easier.

### CODE REVIEW

**Comment by danfulton** [comment]
One caveat is right now any instance of `podman-hpc` will use the same database dir under `/tmp` so if you install two versions, location of `fuse_overlayfs_wrap` will get cached by the first.  I think this is only problematic if you delete the first python environment, because then subsequent calls cannot find the script.

I think the best fix for this is just creating separate dbs depending on python prefix, but I wanted to do this in the `click` branch, after merging these changes.

**Comment by danfulton** [comment]
Oops, forgot I also need to figure out how to get python install process to reliably build `exec-wait` binary, so I'll mark this as a draft in the meantime.

**Comment by danfulton** [comment]
> Oops, forgot I also need to figure out how to get python install process to reliably build `exec-wait` binary, so I'll mark this as a draft in the meantime.

Fixed now with latest commit, and verified the `exec-wait` binary gets build with both RPM build using system python (`/usr/bin/python3.6`) as well as `python -m build` or `pip install .` using `python3.11` in a conda env.

**Comment by danfulton** [comment]
I also did some other cleanup to make sure the python sdist includes everything the RPM build needs.  Perhaps we can just use the python sdist for the RPM source rather than tarballing it manually.

**Comment by lastephey** [comment]
I think it looks generally good. Have you tested the hooks it builds with podman?

**Comment by danfulton** [comment]
> I think it looks generally good. Have you tested the hooks it builds with podman?

I started an ubuntu container and verified the --gpu flag works and copy/bind entries from the yaml show up inside the container, but I didn't run any batch jobs.

**Comment by danfulton** [comment]
Basically a `nvidia-smi` works test.

**Comment by lastephey** [comment]
Ok great. Can you try a 2-node mpich test, too? You can use `registry.nersc.gov/library/nersc/mpi4py:3.1.3`. An easy test is

```
srun -n 2 python3 -m mpi4py.bench helloworld
```

and make sure that both ranks look ok?

**Review by lastephey — CHANGES_REQUESTED** [review]

**Comment by lastephey** [comment]
Sorry, the comments I made last night stayed in 'pending' since I forgot to formally complete my review.

File: setup.cfg [hunk 1]
```diff
@@ -2,14 +2,26 @@
...
-    toml
-    pyyaml
-packages =
-    podman_hpc
+[options.data_files]
+    bin =
+      bin/exec-wait
+    etc/podman_hpc =
+      etc/01-gpu.conf
+      etc/01-mpich.conf
```
[the remark this answers is not in the document]
Comment (danfulton) [reply setup.cfg:23]: The file in the repo is `etc/01.mpich.conf`, so just replicating that.  I think it was changed at some point.  I guess this indicates these files have the same precedence for loading.

File: README.md [hunk 2]
```diff
@@ -2,3 +2,12 @@
 
 Future home of infrastructure developed to help
 the Podman container ecosystem run at NERSC.
+
+## conda env install
```
[the remark this answers is not in the document]
Comment (danfulton) [reply README.md]: Updated `README.md` to clarify RPM build and install vs conda/venv install.

I guess I'm sort of assuming the RPM stuff is still the main install mode, and this env stuff is just for us to develop with, or user installs at other sites that might not have a site wide installation.

[the remark this answers is not in the document]
Comment (danfulton) [reply README.md]: https://github.com/NERSC/podman-hpc/tree/dfulton/packaging#podman-hpc

**Comment by lastephey** [comment]
I tested with 

```
stephey@perlmutter:login30:/pscratch/sd/s/stephey> git clone https://github.com/NERSC/podman-hpc/
Cloning into 'podman-hpc'...
remote: Enumerating objects: 355, done.
remote: Counting objects: 100% (95/95), done.
remote: Compressing objects: 100% (73/73), done.
remote: Total 355 (delta 34), reused 61 (delta 21), pack-reused 260
Receiving objects: 100% (355/355), 3.43 MiB | 7.17 MiB/s, done.
Resolving deltas: 100% (180/180), done.
stephey@perlmutter:login30:/pscratch/sd/s/stephey> pip install .
-bash: pip: command not found
stephey@perlmutter:login30:/pscratch/sd/s/stephey> module load python
stephey@perlmutter:login30:/pscratch/sd/s/stephey> pip install .
Defaulting to user installation because normal site-packages is not writeable
ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.
stephey@perlmutter:login30:/pscratch/sd/s/stephey> 
```

Should the user need to pre-create a custom conda env first?

**Review by scanon — APPROVED** [review]
LGTM

**Review by lastephey — APPROVED** [review]

**Pull request merged**